spec/support/context_helper.cr

```diff
@@ -15,6 +15,56 @@ module ContextHelper
     HTTP::Server::Context.new request, response
   end
 
+  private def build_multipart_request(
+    form_parts : Hash(String, String | Hash(String, String)) = {} of String => String,
+    file_parts : Hash(String, String | Hash(String, String)) = {} of String => String
+  )
+    form_io, content_type = IO::Memory.new, ""
+    HTTP::FormData.build(form_io) do |formdata|
+      content_type = formdata.content_type
+      form_parts.each do |key, value|
+        multipart_form_part(formdata, key, value)
+      end
+      file_parts.each do |key, value|
+        multipart_file_part(formdata, key, value)
+      end
+    end
+    build_request(method: "POST", body: form_io.to_s, content_type: content_type)
+  end
+
+  private def multipart_form_part(
+    formdata : HTTP::FormData::Builder, name : String, value : String
+  )
+    formdata.field(name, value)
+  end
+
+  private def multipart_form_part(
+    formdata : HTTP::FormData::Builder, name : String, value : Hash(String, String)
+  )
+    value.each do |key, nested_value|
+      nested_name = name + ":" + key
+      multipart_form_part(formdata, nested_name, nested_value)
+    end
+  end
+
+  private def multipart_file_part(
+    formdata : HTTP::FormData::Builder, name : String, value : String
+  )
+    file_io = IO::Memory.new(value)
+    metadata = HTTP::FormData::FileMetadata.new(filename: name)
+    headers = HTTP::Headers{"Content-Type" => "text/plain"}
+    formdata.file(name, file_io, metadata, headers)
+  end
+
+  private def multipart_file_part(
+    formdata : HTTP::FormData::Builder, name : String, value : Hash(String, String)
+  )
+    value.each do |key, nested_value|
+      nested_name = name + ":" + key
+      multipart_file_part(formdata, nested_name, nested_value)
+    end
+  end
```

Comment:
I like all the test helpers you wrote 👍

Reply:
Worth extracting to a `multipart_helper.cr`?

Reply:
Yeah I think that's a great idea!